Sources/Sentry/SentryScope.m

```diff
@@ -626,7 +641,12 @@ - (void)addObserver:(id<SentryScopeObserver>)observer
 - (NSDictionary *)buildTraceContext:(nullable id<SentrySpan>)span
 {
     if (span != nil) {
-        return [span serialize];
+        NSDictionary *_Nullable serialized = [span serialize];
+        if (serialized != nil) {
+            return (NSDictionary *_Nonnull)serialized;
+        } else {
+            return @{};
+        }
     } else {
         return [self.propagationContext traceContextForEvent];
     }
```

Comment:
It might make sense to fallback to the `[self.propagationContext traceContextForEvent];`